src/state/user/hooks/index.tsx

```diff
@@ -153,6 +154,26 @@ export function useRemoveUserAddedToken(): (chainId: number, address: string) =>
   )
 }
 
+export function useGasPrice(): string {
```

Comment:
Not sure if we want this to be a hook, since a lot of calls are done outside hooks.
You could use `createSelector` to get rid of useSelector

Reply:
Yea I have managed to get around that by passing the gas price (coming from the hook) as an argument into any function that requires it, but agree it would be nicer not to do that. 

I've added a static function `getGasPrice` in this commit -  see what you think. https://github.com/pancakeswap/pancake-frontend/pull/1857/commits/9ac0844049de7b5586af9e590eff0d5fba357870 `createSelector` requires the state object to be passed to it, so I've opted to access it via the store.

An example of how `getGasPrice` could be used directly outwith a component/hook - https://github.com/pancakeswap/pancake-frontend/pull/1857/commits/9ac0844049de7b5586af9e590eff0d5fba357870#diff-a1efe68899d6806e80fa82e3fcab1c1f16fb7f68b0ece5b34f911196c5842541R18

I do think we still need the hook in order to get an updating value as part of the `useGasPriceManager` hook (used by the settings modal).

Reply:
If we're happy with this approach then I'll implement `getGasPrice` wherever the gas price is accessed outwith a component.